src/construction.cpp

```diff
@@ -938,6 +939,31 @@ void construct::done_vehicle( const tripoint &p )
     g->m.add_vehicle_to_cache( veh );
 }
 
+void construct::done_powered_furniture( const tripoint &p )
```

Comment:
This looks like a simple copy of `done_vehicle`. You should put the common code into a separate function and call that from both `done_` functions.